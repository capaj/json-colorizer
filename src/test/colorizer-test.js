const chai = require('chai')
const { jestSnapshotPlugin } = require('mocha-chai-jest-snapshot')

chai.use(jestSnapshotPlugin())
const { expect } = chai
const { getTokens } = require('../lib/lexer')
const { colorize } = require('../lib/colorizer')
const customColors = {
  BRACE: 'white',
  BRACKET: 'white',
  COLON: 'white',
  COMMA: 'white',
  STRING_KEY: 'yellow',
  NULL_LITERAL: 'red',
  STRING_LITERAL: 'green',
  NUMBER_LITERAL: 'magentaBright',
  BOOLEAN_LITERAL: 'cyan'
}

const fixture = {
  foo: null,
  bar: { baz: true },
  number: 13,
  array: ['values']
}

describe('Colorizer', function () {
  it('does not throw an error when there is whitespace', function () {
    expect(() => colorize(getTokens(JSON.stringify(fixture, null, 2)))).to.not.throw()
  })

  it('colorizes with default options', function () {
    const tokens = getTokens(fixture)
    const result = colorize(tokens)

    expect(result).toMatchSnapshot()
  })

  it('colorizes with custom colors', function () {
    const tokens = getTokens(fixture)
    const result = colorize(tokens, { colors: customColors })

    expect(result).toMatchSnapshot()
  })

  it('colorizes with only specific overrides for colors', function () {
    const tokens = getTokens(fixture)
    const result = colorize(tokens, { colors: { NUMBER_LITERAL: 'red' } })

    expect(result).toMatchSnapshot()
  })
})
