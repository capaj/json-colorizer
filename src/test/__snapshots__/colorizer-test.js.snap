// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colorizer colorizes with custom colors 1`] = `"[37m{[39m[33m\\"foo\\"[39m[37m:[39m[31mnull[39m[37m,[39m[33m\\"bar\\"[39m[37m:[39m[37m{[39m[33m\\"baz\\"[39m[37m:[39m[36mtrue[39m[37m}[39m[37m,[39m[33m\\"number\\"[39m[37m:[39m[95m13[39m[37m,[39m[33m\\"array\\"[39m[37m:[39m[37m[[39m[32m\\"values\\"[39m[37m][39m[37m}[39m"`;

exports[`Colorizer colorizes with default options 1`] = `"[92m{[39m[91m\\"foo\\"[39m[90m:[39m[37mnull[39m[90m,[39m[91m\\"bar\\"[39m[90m:[39m[92m{[39m[91m\\"baz\\"[39m[90m:[39m[36mtrue[39m[92m}[39m[90m,[39m[91m\\"number\\"[39m[90m:[39m[33m13[39m[90m,[39m[91m\\"array\\"[39m[90m:[39m[94m[[39m[92m\\"values\\"[39m[94m][39m[92m}[39m"`;

exports[`Colorizer colorizes with only specific overrides for colors 1`] = `"[92m{[39m[91m\\"foo\\"[39m[90m:[39m[37mnull[39m[90m,[39m[91m\\"bar\\"[39m[90m:[39m[92m{[39m[91m\\"baz\\"[39m[90m:[39m[36mtrue[39m[92m}[39m[90m,[39m[91m\\"number\\"[39m[90m:[39m[31m13[39m[90m,[39m[91m\\"array\\"[39m[90m:[39m[94m[[39m[92m\\"values\\"[39m[94m][39m[92m}[39m"`;
